pkg/data/management/role_data.go

```diff
@@ -618,6 +624,28 @@ func BootstrapAdmin(management *wrangler.Context) (string, error) {
 				logrus.Info("Created default admin user and binding")
 			}
 		}
+	} else {
+		// if we have at least one user, grab the admin user to ensure that it has the Principal label
+		var adminUser *v3.User
+		for _, user := range users.Items {
+			// we could re-list using a label selector, but since we already have a full listing, just use that instead
+			if value, ok := user.Labels[defaultAdminLabelKey]; ok {
+				if value == defaultAdminLabelValue {
+					adminUser = &user
+					break
+				}
+			}
+		}
+		if adminUser == nil {
+			logrus.Errorf("Have more than one user, but unable to locate admin user to adjust labels")
```

Comment:
If Rancher can stumble on after this, I would demote this to logrus.Warnf. If this is critical (it seems like it is?), then it should return an error here as below.

Reply:
So In this case, the failure to find the admin user only inhibits our ability to add the label that we want. I think that returning an error here (effectively locking the user out on upgrade) would be a bit harsh.

Reply:
I'll leave this open until we resolve the other discussion along these lines - but I'll keep parity between this part and the above part (i.e. they will work the same way).